api/docker.ml

```diff
@@ -94,14 +95,12 @@ module Spec = struct
     let include_git = R.include_git_get r in
     let options = { build_args; squash; buildkit; include_git } in
     let push_to =
-      match target, user, password with
-      | "", "", "" -> None
-      | "", _, _
-      | _, "", _
-      | _, _, "" -> Fmt.failwith "push-target, push-user and push-password must be given (or not given) together"
-      | target, user, password ->
+      if user = "" && password <> "" || user <> "" && password = "" then
```

Comment:
This could benefit from some parentheses to make it easier to read. Or maybe:

```suggestion
      if (user = "") <> (password = "") then
```

Reply:
Much neater, I agree!